### DISCUSSION TITLE
setValue on a formValue array without calling useFieldArray

### DISCUSSION THREAD

**Discussion by wdfinch** [react-hook-form/react-hook-form#5397] [Q&A] [answered]
Hi there, I'm wondering what my best option is in this situation. My issue is I want to be able to use `setValue` and change the value of a field array without calling `useFieldArray`. Currently, this doesn't work.

The reason to do this is I have a separate child component that renders the field array in a wizard form. Therefore, if I open the form, `useFieldArray` may not be called and that form child may not be rendered yet. I don't want to use `display: none` for performance reasons. 

Despite this `useFieldArray` not being called, I still need to be able to change my form values including the array values. So far I've found two workarounds:

1. Call `useFieldArray` twice. Once next to `useForm` in the parent and once in my component that actually renders the field array. 

2. Use `reset` rather than `setValue` to change `formValues`

I see two other possible solutions here:

1. We change the behavior so setValue and change array values without useFieldArray having the mount.

2. We look into useFieldArrayContext so I can call useFieldArray at the root level and share the methods throughout my child components.

I personally like one because I think you should be able to use `setValue` and change `formValues` to whatever you want. I know it hasn't always worked like that, but it seems that we have been moving more in this direction with recent updates.

Attached is an example to show the problem. 

https://codesandbox.io/s/goofy-burnell-mqywq?file=/src/App.js

**Comment by bluebill1049** [comment, marked as answer]
This will be difficult because the array field has a special signature, there is no way to determine a normal input (as input value can be any shape) compare with FieldArray.

**Reply by wdfinch** [reply]
I see. Is it safe to call `useFieldArray` twice then? So once in a parent and again later in a child component when I need to mount the field array? 

Also, this could be a use case for useFieldArrayContext. Any ideas there?

**Reply by bluebill1049** [reply]
It's safe to call twice.

**Reply by wdfinch** [reply]
Cool thanks!
